### PR TITLE
Validator Uptime optimizations

### PR DESCRIPTION
**Pull request by krzlabrdx**

### CODE REVIEW

File: src/RadixDlt.NetworkGateway.PostgresIntegration/Models/ValidatorCumulativeEmissionHistory.cs [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+    public long EpochNumber { get; set; }
+
+    [Column("proposals_made")]
+    public long ProposalsMade { get; set; }
+
+    [Column("proposals_missed")]
+    public long ProposalsMissed { get; set; }
+
+    [Column("participation_in_active_set")]
+    public long ParticipationInActiveSet { get; set; }
```
Comment (PawelPawelec-RDX) [inline src/RadixDlt.NetworkGateway.PostgresIntegration/Models/ValidatorCumulativeEmissionHistory.cs:93]: it's not clear what it holds, but i can't figure out better name. Maybe same as in our gateway response `EpochsActiveIn` or `EpochsAsActiveSetMember`

File: src/RadixDlt.NetworkGateway.PostgresIntegration/Models/ValidatorEmissionHistory.cs [hunk 2]
```diff
@@ -67,8 +67,8 @@
 
 namespace RadixDlt.NetworkGateway.PostgresIntegration.Models;
 
-[Table("validator_emission_statistics")]
-internal class ValidatorEmissionStatistics
+[Table("validator_emission_history")]
+internal class ValidatorEmissionHistory
```
Comment (PawelPawelec-RDX) [inline src/RadixDlt.NetworkGateway.PostgresIntegration/Models/ValidatorEmissionHistory.cs:71]: is that table used anywhere after introducing cumulative table with sum for each epoch?

File: src/RadixDlt.NetworkGateway.PostgresIntegration/Models/ValidatorCumulativeEmissionHistory.cs [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+    public long FromStateVersion { get; set; }
+
+    [Column("validator_entity_id")]
+    public long ValidatorEntityId { get; set; }
+
+    [Column("epoch_number")]
+    public long EpochNumber { get; set; }
+
+    [Column("proposals_made")]
+    public long ProposalsMade { get; set; }
```
Comment (PawelPawelec-RDX) [inline src/RadixDlt.NetworkGateway.PostgresIntegration/Models/ValidatorCumulativeEmissionHistory.cs:87]: what do you think about renaming ProposalsMade and ProposalsMissed to sthn that would clearly indicate it's culumative/total count of proposals? i.e `CumulativeProposalsMade`

File: src/RadixDlt.NetworkGateway.PostgresIntegration/Services/ValidatorQuerier.cs [hunk 4]
```diff
@@ -101,26 +96,38 @@ public ValidatorQuerier(ReadOnlyDbContext dbContext, IDapperWrapper dapperWrappe
...
-                epochFrom = fromLedgerState?.Epoch ?? 0,
-            },
-            cancellationToken: token);
+        var validatorIds = validators.Keys.ToList();
+        var epochFrom = fromLedgerState?.Epoch ?? 0;
+        var epochTo = ledgerState.Epoch;
 
-        var validatorUptime = (await _dapperWrapper.QueryAsync<ValidatorUptimeViewModel>(_dbContext.Database.GetDbConnection(), cd, "GetValidatorEmissionStatistics"))
-            .ToDictionary(e => e.ValidatorEntityId);
+        var validatorUptime = await _dbContext
```
Comment (PawelPawelec-RDX) [inline src/RadixDlt.NetworkGateway.PostgresIntegration/Services/ValidatorQuerier.cs:103]: you're missing metric annotation `AnnotateMetricName` I think we've missed these in account locker queries as well.

File: src/RadixDlt.NetworkGateway.PostgresIntegration/Models/ValidatorCumulativeEmissionHistory.cs (hunk 1 above)
Comment (krzlabrdx) [reply src/RadixDlt.NetworkGateway.PostgresIntegration/Models/ValidatorCumulativeEmissionHistory.cs:93]: I think it is very clear. It represents the cumulative sum (see table name) of all participation in active sets.

File: src/RadixDlt.NetworkGateway.PostgresIntegration/Models/ValidatorEmissionHistory.cs (hunk 2 above)
Comment (krzlabrdx) [reply src/RadixDlt.NetworkGateway.PostgresIntegration/Models/ValidatorEmissionHistory.cs:71]: No, it is not. I've left it to reduce the impact of the change (backwards compatibility).

File: src/RadixDlt.NetworkGateway.PostgresIntegration/Models/ValidatorCumulativeEmissionHistory.cs (hunk 3 above)
Comment (krzlabrdx) [reply src/RadixDlt.NetworkGateway.PostgresIntegration/Models/ValidatorCumulativeEmissionHistory.cs:87]: The table itself is already named "cumulative history", I think there's enough context here.

File: src/RadixDlt.NetworkGateway.PostgresIntegration/Services/ValidatorQuerier.cs (hunk 4 above)
Comment (krzlabrdx) [reply src/RadixDlt.NetworkGateway.PostgresIntegration/Services/ValidatorQuerier.cs:103]: Good catch! Added.

File: src/RadixDlt.NetworkGateway.PostgresIntegration/Models/ValidatorEmissionHistory.cs (hunk 2 above)
Comment (PawelPawelec-RDX) [reply src/RadixDlt.NetworkGateway.PostgresIntegration/Models/ValidatorEmissionHistory.cs:71]: i'd vote for removing it but I'm aware we don't have any tool to check if anyone is using that.

File: src/RadixDlt.NetworkGateway.PostgresIntegration/Models/ValidatorCumulativeEmissionHistory.cs (hunk 1 above)
Comment (PawelPawelec-RDX) [reply src/RadixDlt.NetworkGateway.PostgresIntegration/Models/ValidatorCumulativeEmissionHistory.cs:93]: wasn't for me, I had to analyze code to understand what it is. But if you insist we can leave it as it is.

File: src/RadixDlt.NetworkGateway.PostgresIntegration/Models/ValidatorEmissionHistory.cs (hunk 2 above)
Comment (krzlabrdx) [reply src/RadixDlt.NetworkGateway.PostgresIntegration/Models/ValidatorEmissionHistory.cs]: OK, dropped.

File: src/RadixDlt.NetworkGateway.PostgresIntegration/Models/ValidatorCumulativeEmissionHistory.cs (hunk 1 above)
Comment (krzlabrdx) [reply src/RadixDlt.NetworkGateway.PostgresIntegration/Models/ValidatorCumulativeEmissionHistory.cs:93]: I'll leave it as it is.

**Review by PawelPawelec-RDX — APPROVED** [review]

**Pull request merged**